### ISSUE TITLE
Map legend: include nodata field into legend?

### ISSUE THREAD

**Issue by traveller195** [colouring-cities/colouring-core#1091]
**Issue**
We got the feedback, that it would be nice to also have a legend field for 'nodata' values of the visualized building feature.
Otherwise the user could try to compare the buildings with nodata values with similar blue/ gray colours in the map legend, which is not the way it should be interpreted.
Also from a cartographic perspective, it is nice to show all uses map styles in the legend, too.

![grafik](https://user-images.githubusercontent.com/57902113/219403983-e18bdb17-26bd-40a8-b20f-938f442dc2c1.png)

**Describe the solution**
Extend the map legend and include the map style for nodata buildings as an additional field (in the bottom of the legend e.g.)

**Describe any alternatives**
it is a nice to have, could also stay in the current state

**Additional context**
Perhaps, there was also a discussion in the past? but I could not find any related issues

**Comment by polly64** [comment]
@traveller195 yes this sounds good. @mdsimpson42 and @matkoniecz what are you thoughts?

**Comment by matkoniecz** [comment]
Makes sense, but one tricky issue is that it is different in night and day styles. So would require a bit deeper changes, hopefully without duplicating too much of definitions.

Is "No data" fitting all of them? Probably no and should be added only to some.

Is "No data" fitting all legends where it should be shown?

**Comment by traveller195** [comment]
@matkoniecz @polly64 
thanks for your responses.

Oh yes. That´s truth. The map style changes from night to day style. But maybe it is relevant only for the two colours of nodata values (for stroke and fill), which should be different 

**regarding to the current Colouring Core version, it affects the following data categories:**
- Land Use
- Typology (? --> if not all values are automatically preprocessed)
- Size & Form / Size
- Construction
- Age & History
- Street Context (?)
- Planning controls (?) --> visualized builiding attributes can be chosen from a dropdown list
- Energy Perfomance
- Resilience (coming soon)
- Community: for all visualized likes, a map legend field for zero is missing. For public ownership, also nodata could be added.

**except** (because percentage of collected data is visualized):
- Location
- Team

**Comment by matkoniecz** [comment]
> But maybe it is relevant only for the two colours of nodata values (for stroke and fill), which should be different

Yes, but it means that we cannot just add them to current legend data as it accepts single colour and does not support showing different one for day and night style (it would need to be implemented and structure of data changed)

See https://github.com/colouring-cities/colouring-core/blob/master/app/src/frontend/config/category-maps-config.ts
